src/datamodule.py

```diff
@@ -1,33 +1,33 @@
 """
-LightningDataModule to loads Earth Observation data from <file format> using
-<library>.
+LightningDataModule to load Earth Observation data from GeoTIFF files using
+rasterio.
 """
 import lightning as L
+import numpy as np
+import rasterio
 import torch
+import torchdata
 
 
 # %%
-class RandomDataset(torch.utils.data.Dataset):
+def _array_to_torch(filepath: str) -> torch.Tensor:
     """
-    Torch Dataset that returns tensors of size (13, 256, 256) with random
-    values.
+    Read a GeoTIFF file using rasterio into a numpy.ndarray, and convert it
+    to a torch.Tensor (float16 dtype).
     """
+    # GeoTIFF - Rasterio
+    with rasterio.open(fp=filepath) as dataset:
+        array: np.ndarray = dataset.read()
+        tensor: torch.Tensor = torch.as_tensor(data=array.astype(dtype="float16"))
```

Comment:
Is float32 to float16 a save tansformation?

Reply:
There will be some loss of floating point precision, but we'll likely be using 16-bit precision training (see https://lightning.ai/docs/pytorch/2.1.0/common/precision_intermediate.html) to speed up the model training, so best to pre-emptively convert the data to float16 dtype here.